File: lib/strategy-registry.js
import registry from '../data/strategy_registry.json' with {type:'json'};
import blocks from '../data/hard_blocks.json' with {type:'json'};
export function registrySnapshot() {
  return {version:registry.version,strategies:Object.fromEntries(Object.entries(registry.strategies).map(([id,s])=>[id,{...s,...(blocks.blocked[id]?{status:'blocked',...blocks.blocked[id]}:{})}]))};
}

File: scripts/strategy_registry.py
"""Immutable startup snapshot: execution and UI share configuration and block state."""
import json
from pathlib import Path

def registry_snapshot():
    folder=Path(__file__).resolve().parent.parent/'data'
    registry=json.loads((folder/'strategy_registry.json').read_text(encoding='utf-8'))
    blocks=json.loads((folder/'hard_blocks.json').read_text(encoding='utf-8'))['blocked']
    for key,item in registry['strategies'].items():
        if key in blocks:
            item.update(status='blocked',**blocks[key])
    return registry

SNAPSHOT=registry_snapshot()

def enabled(strategy_id):
    return SNAPSHOT['strategies'].get(strategy_id,{}).get('status')=='eligible'
